### ISSUE TITLE
`rb_syserr_fail_str` is declared but undefined.

### ISSUE THREAD

**Issue by nobu** [oracle/truffleruby#3732]
```
have_func: checking for rb_syserr_fail_str(0, Qnil)... -------------------- no

LD_LIBRARY_PATH=. "gcc -o conftest -I/home/runner/.rubies/truffleruby-24.1.1/lib/cext/include -I/home/runner/.rubies/truffleruby-24.1.1/lib/cext/include/ruby/backward -I/home/runner/.rubies/truffleruby-24.1.1/lib/cext/include -I../../../../../ext/io/console    -O3 -fno-fast-math -ggdb3 -Werror=implicit-function-declaration -Wno-int-conversion -Wno-int-to-pointer-cast -Wno-incompatible-pointer-types -Wno-format-extra-args conftest.c  -L.          -L/home/runner/.rubies/truffleruby-24.1.1/lib/cext -Wl,-rpath,/home/runner/.rubies/truffleruby-24.1.1/lib/cext -ltrufflerubytrampoline"
/usr/bin/ld: /tmp/cc266qVm.o: in function `t':
/home/runner/work/io-console/io-console/tmp/x86_64-linux/io/console/3.2.4/conftest.c:15: undefined reference to `rb_syserr_fail_str'
collect2: error: ld returned 1 exit status
Process failed: #<Process::Status: pid 2000 exit 1>
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11: 
12:   return !!argv[argc];
13: }
14: 
15: int t(void) { rb_syserr_fail_str(0, Qnil); return 0; }
/* end */
```

[Workaround](https://github.com/ruby/io-console/pull/81/files) is possible but not preferable of course.

**Comment by andrykonchin** [comment]
Thank you for the report, we'll look into it.

**Comment by andrykonchin** [comment]
Fixed in bcad72e584375b9c6b79d3fc968deb8650a6d67d
